src/platform/user/profile/vet360/reducers/index.js

```diff
@@ -203,15 +209,48 @@ export default function vet360(state = initialState, action) {
     }
 
     case UPDATE_PROFILE_FORM_FIELD: {
+      // The action gets fired upon initial opening of the edit modal
+      // We only want to capture initialFormFields once, it should not update
+      const initialFormFields = isEmpty(state.initialFormFields)
+        ? state.formFields
+        : state.initialFormFields;
+
       const formFields = {
         ...state.formFields,
         [action.field]: action.newState,
       };
-      return { ...state, formFields };
+
+      const modalName = state?.modal;
+      const initialFormFieldValues = state.initialFormFields[modalName]?.value;
+      let formFieldValues = formFields[modalName]?.value;
+
+      // Initial form fields does not have 'view' properties, those get added to formFields
+      // After editing a field. So we need to strip of those 'view' fields to be able to compare
+      formFieldValues = pickBy(formFieldValues, value => value !== undefined);
+      formFieldValues = pickBy(
+        formFieldValues,
+        (value, key) => !key.startsWith('view:'),
+      );
+
+      const hasUnsavedEdits =
+        !isEmpty(initialFormFields) &&
+        !isEqual(formFieldValues, initialFormFieldValues);
+
+      return {
+        ...state,
+        formFields,
+        hasUnsavedEdits,
+        initialFormFields,
+      };
     }
 
     case OPEN_MODAL:
-      return { ...state, modal: action.modal, modalData: action.modalData };
+      return {
+        ...state,
+        modal: action.modal,
+        modalData: action.modalData,
+        hasUnsavedEdits: false,
```

Comment:
I think we need to set `initialFormFields: {}` here as well. I think that'll fix a bug I found.

The bug:
- start editing home address
- immediately stop editing home address without making changes
- start editing mailing address but don't make any changes
- navigate away and you'll get a warning because it's comparing the current mailing address data to the home address data

Reply:
Good catch, again! Updated 👍